src/protocols/ethereum/libs/EthereumAdapter.ts

```diff
@@ -191,11 +194,124 @@ export class EthereumAdapter extends BaseProtocolAdapter {
   }
 
   override async fetchAvailableTokens(): Promise<IToken[]> {
-    return [DUMMY_ERC20_TOKEN]; // TODO replace with real tokens fetching
+    const { ethActiveNetworkPredefinedSettings } = useEthNetworkSettings();
+    const apiUrl = ethActiveNetworkPredefinedSettings.value.tokenMiddlewareUrl;
+    const res = await fetchJson(`${apiUrl}getTop?apiKey=freekey`);
+    if (res.error) {
+      throw res.error.message;
+    }
+    return res.tokens ? res.tokens.map((token: any) => ({
+      contractId: token.address,
+      decimals: token.decimals || undefined,
+      image: token.image ? `https://ethplorer.io/${token.image}` : undefined,
+      name: token.name,
+      protocol: PROTOCOLS.ethereum,
+      symbol: token.symbol,
+    })) : [];
+  }
+
+  override async fetchAccountTokenBalances(address: string): Promise<ITokenBalance[]> {
+    const { ethActiveNetworkPredefinedSettings } = useEthNetworkSettings();
+    const apiUrl = ethActiveNetworkPredefinedSettings.value.tokenMiddlewareUrl;
+    const res = await fetchJson(`${apiUrl}getAddressInfo/${address}?showETHTotals=false&showTxsCount=false&apiKey=freekey`);
+    if (res.error) {
+      throw res.error.message;
+    }
+    return res.tokens?.map(({
+      rawBalance, tokenInfo: { address: contractId, decimals },
+    }: EthRawToken) => ({
+      address,
+      amount: rawBalance,
+      contractId,
+      convertedBalance: +toShiftedBigNumber(rawBalance, -decimals).toFixed(2),
+      protocol: PROTOCOLS.ethereum,
+    })) || [];
+  }
+
+  override async fetchTokenInfo(contractId: string): Promise<IToken> {
```

Comment:
I suggest to catch errors in the body of this function or eventually in the place where it's executed.